beater/middleware/authorization_middleware.go

```diff
@@ -41,19 +41,19 @@ func RequireAuthorizationMiddleware(token string) Middleware {
 
 			c.Authorized = true
 			h(c)
-		}
+		}, nil
```

Comment:
I disagree, and prefer to use state at least as possible. Also, if I understand correctly, if I were to do that, I would be returning an error on each request instead of once during initialization.

Reply:
see https://github.com/elastic/apm-server/pull/2706#discussion_r324140270